core/src/main/scala/com/microsoft/azure/synapse/ml/codegen/CodeGen.scala

```diff
@@ -134,11 +135,27 @@ object CodeGen {
   }
 
   //noinspection ScalaStyle
+  //scalastyle:off
   def generatePyPackageData(conf: CodegenConfig): Unit = {
     if (!conf.pySrcDir.exists()) {
       conf.pySrcDir.mkdir()
     }
     val extraPackage = if (conf.name.endsWith("core")){" + [\"mmlspark\"]"}else{""}
+    val requireList = if(conf.name.contains("deep-learning")) {
+      s"""MINIMUM_SUPPORTED_PYTHON_VERSION = "3.8"
+         |dl_require_list = [
+         |    "cmake",
+         |    "torch==1.11.0",
+         |    "pytorch_lightning>=1.5.0,<1.5.10",
+         |]
+         |
+         |dl_extra_list = ["torchvision>=0.12.0", "horovod==0.25.0"]""".stripMargin
+    } else ""
+    val extraRequirements = if (conf.name.contains("deep-learning")) {
+      s"""    install_requires=dl_require_list,
+         |    extras_require={"extras": dl_extra_list},
+         |    python_requires=f">={MINIMUM_SUPPORTED_PYTHON_VERSION}",""".stripMargin
+    } else ""
```

Comment:
nit: can this just be extraRequirements as opposed to also the dependency list? seems like its just additional stuff to override

Reply:
If its a matter of indentation theres a utility methoid for this

Reply:
Done.